webapp/tests/karma/ts/modules/contacts/contacts.component.spec.ts

```diff
@@ -152,74 +183,60 @@ describe('Contacts component', () => {
-  //         create_form: 'form:contact:create:type3',
-  //       },
-  //     ]);
-  //     const forms = [
-  //       { _id: 'form:contact:create:type3' },
-  //       { _id: 'form:contact:create:type2' },
-  //     ];
-  //     xmlForms.listen.callsArgWith(2, null, forms);
-
-  //     const ctrl = createController();
-  //     return ctrl
-  //       .getSetupPromiseForTesting()
-  //       .then(() => {
-  //         assert.equal(xmlForms.listen.callCount, 1);
-  //         assert.deepEqual(xmlForms.listen.args[0][1], { contactForms: true });
-  //         assert.deepEqual(ctrl.setLeftActionBar.args[0][0].childPlaces, [
-  //           {
-  //             id: 'type2',
-  //             create_form: 'form:contact:create:type2',
-  //           },
-  //           {
-  //             id: 'type3',
-  //             create_form: 'form:contact:create:type3',
-  //           },
-  //         ]);
-
-  //         //search still searches with all types
-  //         assert.deepEqual(searchService.args[0][1], { types: { selected: ['type1', 'type2', 'type3'] } });
-  //       });
-  //   });
-
-  // });
+  describe('Action bar', () => {
+    it('should initialise action bar', fakeAsync(() => {
+      flush();
+
+      expect(globalActions.setLeftActionBar.callCount).to.equal(1);
+      expect(globalActions.setLeftActionBar.args[0][0].childPlaces.length).to.equal(0);
+      expect(globalActions.setLeftActionBar.args[0][0].hasResults).to.equal(false);
+      expect(globalActions.setLeftActionBar.args[0][0].userFacilityId).to.equal('district-id');
+      expect(globalActions.setLeftActionBar.args[0][0].exportFn).to.be.a('function');
+    }));
+
+    it('should filter contact types to allowed ones from all contact forms', fakeAsync(() => {
+      sinon.resetHistory();
+      contactTypesService.getChildren.resolves([
+        {
+          id: 'type1',
+          create_form: 'form:contact:create:type1',
+        },
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
+      const forms = [
+        { _id: 'form:contact:create:type3' },
+        { _id: 'form:contact:create:type2' },
+      ];
+
+      component.ngOnInit();
+      flush();
+
+      expect(xmlFormsService.subscribe.callCount).to.equal(1);
+      expect(xmlFormsService.subscribe.args[0][0]).to.equal('ContactForms');
+      expect(xmlFormsService.subscribe.args[0][1]).to.deep.equal({ contactForms: true });
+
+      xmlFormsService.subscribe.args[0][2](null, forms);
+
+      expect(globalActions.updateLeftActionBar.callCount).to.equal(1);
+      expect(globalActions.updateLeftActionBar.args[0][0].childPlaces).to.have.deep.members([
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
```

Comment:
I see that an assertion about the parameters passed to the search function has been omitted from this test. 
Is it no longer relevant?

Reply:
Left action bar doesn't do search and the xml forms subscription is only used for getting the allowed forms for left action bar, so not affecting search feature. I think we don't need it.
Am I missing something? 🤔

Reply:
The purpose of the assertion was to check that the function that is now called `filterAllowedChildType` doesn't mutate `this.childPlaces` which is used in search.

Reply:
Ah I see... in the new code, childPlaces is used once to defined the [defaultFilters](https://github.com/medic/cht-core/blob/7c28e25b12a3cb8905d0483a0bbea6f86a9fbf51/webapp/src/ts/modules/contacts/contacts.component.ts#L151) but using a map function which returns a new array. In this case I will check that the content of childPlaces is untouch after the filter.  Thank you!

Reply:
Done, added a comment to it's more evident the why 👍